### PR TITLE
Replaced use of lockfile with fasteners and optimised locking code.

### PR DESCRIPTION
**Pull request by mazimkhan**
Changes:
- Replaced ```lockfile``` with ```fasteners``` 
- Created a decorator for methods that require inter process locking.

@bridadan @adbridge Please review.

### CODE REVIEW

File: mbed_lstools/lstools_base.py [hunk 1]
```diff
@@ -260,13 +285,13 @@ def list_manufacture_ids(self):
 
         return pt.get_string()
 
+    @timed_mbedls_lock(60)
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: Digging the decorator syntax, nice!

File: mbed_lstools/lstools_base.py [hunk 2]
```diff
@@ -323,20 +337,8 @@ def write_mock_file(filename, mock_ids):
...
-        while not lock.i_am_locking():
-            try:
-                lock.acquire(timeout=1)
-            except LockTimeout as e:
-                lock.break_lock()
-                lock.acquire()
-                self.debug(self.mock_read.__name__, str(e))
-        self.debug(self.mock_write.__name__, "locked '%s'"% lock.path)
-
         result = write_mock_file(self.MOCK_HOME_FILE_NAME, mock_ids)
```
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: Any reason why the function `write_mock_file` can't be inlined here? I suppose this was mostly keep it parallel to how `mock_read` is written?

**Comment by mazimkhan** [comment]
Fixes issue #130

File: mbed_lstools/lstools_base.py (hunk 2 above)
Comment (bridadan) [inline mbed_lstools/lstools_base.py]: I'm fine with revisiting this in another PR if we decide to do so, shouldn't block this PR!

**Pull request merged**